src/main/kotlin/com/salesforce/dynamodbv2/mt/backups/MtBackupTableSnapshotter.kt

```diff
@@ -0,0 +1,123 @@
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.model.BackupStatus
+import com.amazonaws.services.dynamodbv2.model.ContinuousBackupsUnavailableException
+import com.amazonaws.services.dynamodbv2.model.CreateBackupRequest
+import com.amazonaws.services.dynamodbv2.model.CreateBackupResult
+import com.amazonaws.services.dynamodbv2.model.DeleteBackupRequest
+import com.amazonaws.services.dynamodbv2.model.DeleteTableRequest
+import com.amazonaws.services.dynamodbv2.model.DescribeBackupRequest
+import com.amazonaws.services.dynamodbv2.model.DescribeBackupResult
+import com.amazonaws.services.dynamodbv2.model.DescribeTableRequest
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput
+import com.amazonaws.services.dynamodbv2.model.RestoreTableFromBackupRequest
+import org.slf4j.LoggerFactory
+
+/**
+ *
+ * Utility to take correct backups of live production data with minimal risk.
+ *
+ * Instead of trying to operate against live tables taking traffic from production, backups operate against snapshotted
+ * tables, to scan through at their own pace. These tables are snapshotted using DynamoDB's own on-demand backups,
+ * which are leverage to generate single point in time snapshots of live data. This protects against any interleaved
+ * writes customers may drive against a live table, given generating S3 tenant-table backups are a time consuming
+ * operation. Additionally, it avoids competing with customer driven IOPs against prod dynamo tables, by getting a
+ * queiesced table that the backup generator can exclusively consume all IOPs available to generate the backup either
+ * as fast as, or as slow as needed.
+ *
+ */
+open class MtBackupTableSnapshotter {
+
+    private val logger = LoggerFactory.getLogger(MtBackupTableSnapshotter::class.java)
+    /**
+     * Take a given dynamo table, and fire consecutive backup and restore requests to snapshot said table to a new
+     * table space location. This is a time consuming operation, and should be done asyncronously, and potentially
+     * resumed if interrupted.
+     */
+    open fun snapshotTableToTarget(snapshotRequest: SnapshotRequest): SnapshotResult {
+        val startTime = System.currentTimeMillis()
+        logger.info("Snapshot of ${snapshotRequest.sourceTableName} to ${snapshotRequest.targetTableName} beginning")
+        // generate backup of table
+        var backupResult: CreateBackupResult? = null
+        do {
+            try {
+                backupResult = snapshotRequest.amazonDynamoDb.createBackup(CreateBackupRequest()
+                        .withBackupName(snapshotRequest.targetTableName)
+                        .withTableName(snapshotRequest.sourceTableName))
+            } catch (e: ContinuousBackupsUnavailableException) {
+                // wait for backups to be ready on this table, if not already
+                Thread.sleep(1000L)
```

Comment:
Could use Awaitility https://github.com/salesforce/mt-dynamo/blob/39911ab376ba8dafed59f0c87d4f85d77168d6a0/src/main/java/com/salesforce/dynamodbv2/mt/admin/AmazonDynamoDbAdminUtils.java#L99

Reply:
makes sense, will include that in followup PR